### PR TITLE
Add scopes

### LINKED ISSUE #3: Integrate scopes
We add scopes for queries in Accentor/web#413
Once that PR is merged we should add those scopes to this client and document their usage.

### PR DESCRIPTION
**Pull request by robbevp**
Fix #3 
Add the same scopes as accentor/web#413

Instead of setting the initial scope as a class field, I've added them in the constructor. This should not make any functional difference, but the class fields syntax is newer and not yet support by the default eslint parser.

I've also changed the export of `createApiClient` from a default export to a named export. It makes more sense to import everything as `import { createApiSync, AlbumsScope, ... } from ...` instead of import createApiSync { AlbumsScope, ... } from ...`. (This is also easier to build to CJS)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ All functions should be provided with an auth object containing the `device_id`
 Not every function is available for every model. Only the routes available to that model can be called.
 ```js
 // Index
-const generator = api.tracks.indexGenerator(auth);
+const generator = api.tracks.indexGenerator(auth, scope?);
```
Comment (chvp) [inline README.md:20]: The `?` is weird here, if this is a code example.

**Review by chvp — CHANGES_REQUESTED** [review]
This isn't well-typed. Every `index` method can take any `Scope` class if you look at the generated `index` methods. I don't have an immediate suggestion on how to fix it, but I do think it should be fixed.

**Comment by robbevp** [comment]
> This isn't well-typed. Every `index` method can take any `Scope` class if you look at the generated `index` methods. I don't have an immediate suggestion on how to fix it, but I do think it should be fixed.

I've updated this, so that we can distinguish between indexes with scopes and without. I had to introduce Babel's eslint-parser to use private fields. Browser support is still limited for this, but esbuild can transform this for us during build

**Review by chvp — COMMENTED** [review]
Hmm, it's still not clear from the API that (for example) the tracks index method only takes a TrackScope.

**Comment by robbevp** [comment]
> Hmm, it's still not clear from the API that (for example) the tracks index method only takes a TrackScope.

I don't think this is that much of an issue. IMHO its the responsibility of the developer to use this correctly. Someone can also pass a string or an object to a scope that expects an ID, or pass the id from another type of object 🤷 
If we switch to typescript, we can provide some more guidance (even in JS projects)

**Review by chvp — APPROVED** [review]

**Pull request merged**